### PR TITLE
feat(mongo): update mongo driver to 6.3.0 (PL-637)

### PR DESCRIPTION
**Pull request by theProf**
**Fixes or implements PL-637**

### Brief description. What is this change?

Update mongo driver from 3.x to 6.3.0
Update node from v16.13 to v20

### Implementation details. How do you make this change?

Update mongo driver from v3 -> 4 -> 5 -> 6 to flesh out issues. We have very rudimentary usage of mongo and no large issues arose from typescript or unit tests. Manual testing is a **MUST**. (possible addition of smoke test?)

### Related PRs

- https://github.com/voiceflow/general-runtime/pull/659

### Checklist

- [ ] Appropriate tests have been written
    - [ ] Add least 1 Smoke Test (?)
    - [x] Existing smoke tests
    - [ ] Manual testing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -190,5 +189,9 @@
     "test:run": "NODE_ENV=test nyc ts-mocha --paths --opts ./config/tests/mocha.opts 'tests/**/*.{unit,it}.ts'",
     "test:single": "NODE_ENV=test ts-mocha --paths --opts ./config/tests/mocha.opts",
     "test:unit": "yarn && NODE_ENV=test nyc --report-dir=nyc_coverage_unit ts-mocha --paths --opts ./config/tests/mocha.opts 'tests/**/*.unit.ts'"
+  },
+  "volta": {
```
Comment (effervescentia) [inline package.json:193]: 👍

File: package.json [hunk 2]
```diff
@@ -129,15 +128,15 @@
...
     "tsconfig-paths": "^4.0.0",
-    "ttypescript": "^1.5.13",
     "typedoc": "^0.20.16",
     "typedoc-plugin-sourcefile-url": "^1.0.6",
     "typescript": "~4.7.2",
     "typescript-transform-paths": "3.3.1"
   },
   "engines": {
-    "node": ">=16"
+    "node": ">=20"
```
Comment (effervescentia) [inline package.json:139]: 👍

File: .yarnrc.yml
Comment (effervescentia) [inline .yarnrc.yml:1]: :thu

File: .circleci/continue_config.yml.tmpl
Comment (effervescentia) [inline .circleci/continue_config.yml.tmpl:1]: I think `vfcommon/release` also needs an updated executor
https://github.com/voiceflow/general-service/blob/master/.circleci/continue_config.yml.tmpl#L97

**Review by effervescentia — COMMENTED** [review]
yarn / node / volta changes look good 👍

**Comment by theProf** [comment]
split the volta, node, yarn, ci changes out into it's own PR: https://github.com/voiceflow/general-runtime/pull/659

**Review by trs — APPROVED** [review]

**Pull request merged**